src/unasync/__init__.py

```diff
@@ -27,6 +37,92 @@
+        self.token_replacements = _ASYNC_TO_SYNC.copy()
+        for key, val in (additional_replacements or {}).items():
+            self.token_replacements[key] = val
+
+    def match(self, filepath):
+        """Determines if a Rule matches a given filepath and if so
+        returns a higher comparable value if the match is more specific.
+        """
+        file_segments = [x for x in filepath.split(os.sep) if x]
+        from_segments = [x for x in self.fromdir.split(os.sep) if x]
+        len_from_segments = len(from_segments)
+
+        if len_from_segments > len(file_segments):
+            return False
+
+        for i in range(len(file_segments) - len_from_segments + 1):
+            if file_segments[i : i + len_from_segments] == from_segments:
+                return len_from_segments, i
+
+        return False
+
+    def unasync_file(self, filepath):
+        with open(filepath, "rb") as f:
+            write_kwargs = {}
+            if sys.version_info[0] >= 3:
+                encoding, _ = std_tokenize.detect_encoding(f.readline)
+                write_kwargs["encoding"] = encoding
+                f.seek(0)
+            tokens = tokenize(f)
+            tokens = self.unasync_tokens(tokens)
+            result = untokenize(tokens)
+            outfilepath = filepath.replace(self.fromdir, self.todir)
+            makedirs_existok(os.path.dirname(outfilepath))
+            with open(outfilepath, "w", **write_kwargs) as f:
+                print(result, file=f, end="")
+
+    def unasync_tokens(self, tokens):
+        # TODO __await__, ...?
+        used_space = None
+        for space, toknum, tokval in tokens:
+            if tokval in ["async", "await"]:
+                # When removing async or await, we want to use the whitespace that
+                # was before async/await before the next token so that
+                # `print(await stuff)` becomes `print(stuff)` and not
+                # `print( stuff)`
+                used_space = space
+            else:
+                if toknum == std_tokenize.NAME:
+                    tokval = self.unasync_name(tokval)
+                elif toknum == std_tokenize.STRING:
+                    left_quote, name, right_quote = tokval[0], tokval[1:-1], tokval[-1]
+                    tokval = left_quote + self.unasync_name(name) + right_quote
+                if used_space is None:
+                    used_space = space
+                yield (used_space, tokval)
+                used_space = None
+
+    def unasync_name(self, name):
+        if name in self.token_replacements:
+            return self.token_replacements[name]
+        # Convert classes prefixed with 'Async' into 'Sync'
+        elif len(name) > 5 and name.startswith("Async") and name[5].isupper():
+            return "Sync" + name[5:]
+        return name
+
+
+_DEFAULT_RULE = Rule(fromdir="/_async/", todir="/_sync/")
+
+unasync_tokens = _DEFAULT_RULE.unasync_tokens
+unasync_name = _DEFAULT_RULE.unasync_name
```

Comment:
nitpick: I don't think we need to worry about the public API here, nobody is using those functions right now